dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/upload/UploadService.java

```diff
@@ -112,33 +114,76 @@ private void startUpload(File file, String objectId, String md5, boolean overwri
       throw new NotResumableException(e);
     }
 
+    // delete if already present
+    if (overwrite) {
+      UploadStateStore.create(file, spec, false);
+    }
+
     val progress = new Progress(terminal, quiet, spec.getParts().size(), 0);
     uploadParts(spec.getParts(), file, objectId, spec.getUploadId(), progress);
+    cleanupState(file, objectId);
   }
 
   /**
    * Resume a upload if it is possible. Otherwise, it will start a new upload. Resume might not be possible if the
    * upload progress cannot be retrieved.
    */
   @SneakyThrows
-  private void resumeIfPossible(File file, String objectId, String md5, boolean checksum) {
+  private void resumeIfPossible(File uploadFile, String objectId, String md5, boolean checksum) {
     UploadProgress progress = null;
     try {
-      progress = storageService.getProgress(objectId, file.length());
+      // progress = storageService.getProgress(objectId, file.length());
+      progress = checkProgress(uploadFile, objectId);
     } catch (NotRetryableException e) {
-      log.info("New upload: {}", objectId);
-      startUpload(file, objectId, md5, true);
+      // org.icgc.dcc.storage.client.exception.ServiceRetryableResponseErrorHandler translates the 404 received from
+      // server into a NotRetryableException
+      log.info("No upload id found for object id {}. Start new upload.", objectId);
+      startUpload(uploadFile, objectId, md5, true);
       return;
     }
-    resume(file, progress, objectId, checksum);
+    resume(uploadFile, progress, objectId, checksum);
+  }
+
+  @SneakyThrows
+  private UploadProgress checkProgress(File uploadFile, String objectId) {
+    /*
+     * see if there is already an upload in progress for this object id fetch upload id if present, send if missing,
+     * send null
+     */
+    val uploadId = UploadStateStore.fetchUploadId(uploadFile, objectId);
+    UploadProgress progress = null;
+    progress = storageService.getProgress(objectId, uploadFile.length());
+
+    // compare upload id's
+    if (uploadId.isPresent() && progress != null) {
+      if (uploadId.get().equalsIgnoreCase(progress.getUploadId())) {
+        // can continue - upload id's match
+      } else {
+        val msg =
```

Comment:
If this is a common pattern throughout the codebase, I would create a reusable utility:

```java
checkResumable(Local in-progress upload %s conflicts with remote upload in progress %s. Aborting local upload.", uploadId.get(), progress.getUploadId())
```

Which will cut down on a lot of boilerplate all over the place and make it easier to read.